test/dynamo/test_misc.py

```diff
@@ -6572,6 +6573,36 @@ def test_recompile(foo, *, exp_frame_count):
         torch._dynamo.reset()
         test_recompile(foo_graph_break, exp_frame_count=2)
 
+    def test_multithread_compile_dynamic(self):
+        @torch.compile(backend="eager", dynamic=False)
```

Comment:
We're implicitly testing `config.patch` here - could we also add a test where we explicitly call patch?